### PR TITLE
fix: added folder object in ListFileResponse

### LINKED ISSUE #105: listFiles response type is not valid for folders
The response type for the `listFIiles` API is only valid for file objects, but if the API is used to fetch folders the response will be different, for example:
```typescript
{
  type: 'folder',
  name: 'folder-name',
  createdAt: '2024-12-02T20:54:59.433Z',
  updatedAt: '2024-12-02T20:54:59.433Z',
  folderId: '674e1ea3e375273f609c',
  folderPath: '/path/to/folder'
}
```

Unfortunately this is not even documented in the [official API documentation](https://imagekit.io/docs/api-reference/digital-asset-management-dam/list-and-search-assets#List-and-search-assets) of ImageKit. And there might even be a third response type for `file-version`. All available types according to the docs are:
- `file`
- `file-version`
- `folder`
- `all`

From the top of my head a solution to this would be making the response type dependent on the input type parameter (with `all` returning a union of all types). This should be quite clean and users can easily narrow down type using `type` in each object.

### PR DESCRIPTION
**Pull request by ahnv**
Closes #105

### CODE REVIEW

File: libs/interfaces/ListFile.ts [hunk 1]
```diff
@@ -76,50 +76,4 @@ export interface ListFileOptions {
  *
  * @see {@link https://docs.imagekit.io/api-reference/media-api/list-and-search-files#response-structure-and-status-code-application-json}
  */
-export interface ListFileResponse {
```
Comment (ahnv) [inline libs/interfaces/ListFile.ts:79]: I have made sure that the ListFileResponse is not being used anywhere else

File: libs/interfaces/Item.ts [hunk 2]
```diff
@@ -1,4 +0,0 @@
-/**
- * Type of returned item. It can be either file or folder.
- */
-export type Item = "file" | "folder";
```
Comment (ahnv) [inline libs/interfaces/Item.ts:4]: This is no longer required

File: libs/interfaces/ListFile.ts (hunk 1 above)
Comment (imagekitio) [reply libs/interfaces/ListFile.ts:79]: But FileObject is being used in many places and it would be incorrect now

File: README.md [hunk 3]
```diff
@@ -27,6 +27,32 @@ ImageKit is complete media storage, optimization, and transformation solution th
 
 ## Changelog
 
+### SDK Version 5.3.0
+
+### Breaking changes
```
Comment (imagekitio) [inline README.md:32]: This is increasing. Create a separate changelog.md and just link in main readme.

File: libs/interfaces/ListFile.ts (hunk 1 above)
Comment (ahnv) [reply libs/interfaces/ListFile.ts:79]: Yes, FileObject is being used at other places, but it's being used in getFileDetails, getFileVersions, getFileVersionDetails, updateFileDetails, restoreFileVersion. 

All the above functions deal with File and FileVersion

**Pull request merged**